examples/gno.land/p/demo/entropy/entropy.gno

```diff
@@ -0,0 +1,85 @@
+// Entropy generates fully deterministic, cost-effective, and hard to guess
+// numbers.
+//
+// It is designed both for for single-usage, like seeding math/rand or for being
+// reused which increases the entropy and its cost effectiveness.
+//
+// Disclaimer: this package is unsafe and won't prevent others to guess values
+// in advance.
+//
+// It uses the Bernstein's hash djb2 to be CPU-cycle efficient.
+package entropy
+
+import (
+	"std"
+	"time"
+)
+
+type Instance struct {
+	value int64
+}
+
+func New() *Instance {
+	r := Instance{value: 5381}
+	r.addEntropy()
+	return &r
+}
+
+func FromSeed(seed int64) *Instance {
+	r := Instance{value: seed}
+	r.addEntropy()
+	return &r
+}
+
+func (i *Instance) Seed() int64 {
+	return i.value
+}
+
+func (i *Instance) djb2String(input string) {
+	for _, c := range input {
+		i.djb2Int64(int64(c))
+	}
+}
+
+// super fast random algorithm.
+// http://www.cse.yorku.ca/~oz/hash.html
+func (i *Instance) djb2Int64(input int64) {
```

Comment:
The original implementation of djb2 was designed for `uint32` integers. Are you sure that using `int64` maintains the same properties? Are the magics `5381` and `33` the right ones here?

Reply:
Not sure about that, let's switch to uint32.